VideoLocker/src/main/java/org/edx/mobile/module/analytics/ISegmentImpl.java

```diff
@@ -540,21 +540,39 @@ public Properties trackUserLogout() {
     }
 
     /**
-     * This function is used send Screen View Details
-     * @param screenName
-     * @return 
+     * This function is used to send Screen View Details
+     *
+     * @param screenName The screen name to track
+     * @return
      */
     @Override
-    public Properties screenViewsTracking(String screenName) {
-        try{
-            SegmentAnalyticsEvent aEvent = new SegmentAnalyticsEvent();
-            aEvent.setAppNameContext();
-            tracker.screen("", screenName, aEvent.properties);
-            return aEvent.properties;
-        }catch(Exception e){
-            logger.error(e);
-            return null;
+    public Properties trackScreenView(String screenName) {
+        return trackScreenView(screenName, null, null);
+    }
+
+    /**
+     * This function is used to send Screen View Details with an extra event of screen view with
+     * more details
+     *
+     * @param screenName The screen name to track
+     * @param courseId   course id of the course we are viewing
+     * @param action     any custom action we need to send with event
+     * @return
+     */
+    @Override
+    public Properties trackScreenView(String screenName, String courseId, String action) {
+        // Sending screen view
+        SegmentAnalyticsEvent aEvent = new SegmentAnalyticsEvent();
+        aEvent.setAppNameContext();
+        if (!TextUtils.isEmpty(action)) {
+            aEvent.properties.put(Keys.ACTION, action);
         }
+        tracker.screen("", screenName, aEvent.properties);
+
+        // Sending screen event
+        addCategoryToBiEvents(aEvent.properties, Values.SCREEN, screenName);
+        aEvent.properties.put(Keys.COURSE_ID, TextUtils.isEmpty(courseId) ? "" : courseId);
```

Comment:
Err. Nevermind the thing I just said was totally wrong. As far as I can tell, in that code I only see ``action`` being sent on the ``screen`` call not with ``trackEvent``? Let me double check this - I think that's backwards, since ``screen`` events don't have extra content